internal/collector/dns/dns.go

```diff
@@ -493,5 +509,52 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) error {
 		c.perfDataObject[0].SecureUpdateReceived,
 	)
 
+	// Query DNS error statistics
+	application, err := mi.ApplicationInitialize()
+	if err != nil {
+		return fmt.Errorf("failed to initialize MI application: %w", err)
+	}
+	defer application.Close()
+
+	session, err := application.NewSession(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create MI session: %w", err)
+	}
+	defer session.Close()
+
+	var stats []DNSStatistic
+	query, err := mi.NewQuery("SELECT Name, CollectionName, Value, DnsServerName FROM MicrosoftDNS_Statistic")
+	if err != nil {
+		return fmt.Errorf("failed to create query: %w", err)
+	}
+
+	err = session.QueryUnmarshal(&stats, mi.OperationFlagsStandardRTTI, nil, mi.NamespaceRootMicrosoftDNS, mi.QueryDialectWQL, query)
+	if err != nil {
+		return fmt.Errorf("failed to query DNS statistics: %w", err)
+	}
+
+	// Collect DNS error statistics
+	seenStats := make(map[string]bool)
+	for _, stat := range stats {
+		// Only collect metrics from Error Stats collection
+		if stat.CollectionName != "Error Stats" {
+			continue
+		}
+
+		// Create a unique key for this combination of labels
+		key := fmt.Sprintf("%s_%s_%s", stat.Name, stat.CollectionName, stat.DnsServerName)
+		if !seenStats[key] {
+			ch <- prometheus.MustNewConstMetric(
+				c.dnsEnhancedStats,
+				prometheus.CounterValue,
+				float64(stat.Value),
+				stat.Name,
+				stat.CollectionName,
+				stat.DnsServerName,
+			)
+			seenStats[key] = true
+		}
+	}
+
```

Comment:
It would be great to move the WMI section to a separate sub collector. 

In history it help me a lot to have 1 datasource per collector. If a collector uses multiple datasources, then split the collector into smaller sub collectors.

One code example is here https://github.com/prometheus-community/windows_exporter/blob/85455bd04dba483894f657f8dc4b4d0286cc39de/internal/collector/net/net.go#L298-L314 where the `Collect` method calling 2 "sub" collector methods.

one name could be `collectPerfData`, the second one could be `collectWMI`. At least this pattern is used across the whole exporter.